test/pkg/firewall/firewall.go

```diff
@@ -0,0 +1,165 @@
+package firewall
+
+import (
+	"fmt"
+	"log"
+	"os"
+	"path/filepath"
+
+	v1 "k8s.io/api/core/v1"
+
+	"github.com/openshift-kni/commatrix/pkg/consts"
+	"github.com/openshift-kni/commatrix/pkg/utils"
+)
+
+// Apply the firewall rules on the node.
+func ApplyRulesToNode(NFTtable []byte, nodeName string, utilsHelpers utils.UtilsInterface) error {
+	debugPod, err := utilsHelpers.CreatePodOnNode(nodeName, consts.DefaultDebugNamespace, consts.DefaultDebugPodImage)
+	if err != nil {
+		return fmt.Errorf("failed to create debug pod on node %s: %w", nodeName, err)
+	}
+
+	defer func() {
+		err := utilsHelpers.DeletePod(debugPod)
+		if err != nil {
+			log.Printf("failed cleaning debug pod %s: %v", debugPod, err)
+		}
+	}()
+
+	// save the nftabes on  /host/etc/nftables/firewall.nft
+	_, err = runBashCommandOnPod(debugPod, fmt.Sprintf("echo '%s' > /host/etc/nftables/firewall.nft", string(NFTtable)), false, utilsHelpers)
+	if err != nil {
+		return fmt.Errorf("failed to save rule set to file on node %s: %w", nodeName, err)
+	}
+
+	// run apply nft command for the rules
+	_, err = runBashCommandOnPod(debugPod, "nft -f /etc/nftables/firewall.nft", true, utilsHelpers)
+	if err != nil {
+		return fmt.Errorf("failed to apply rule set on node %s: %w", nodeName, err)
+	}
+
+	output, err := nftList(debugPod, utilsHelpers)
+	if err != nil {
+		return err
+	}
+
+	// edit the sys conf to make sure it will be on the list after reboot
+	err = editNftablesConf(debugPod, utilsHelpers)
+	if err != nil {
+		return err
+	}
+
+	_, err = runBashCommandOnPod(debugPod, "/usr/sbin/nft list ruleset > /etc/nftables.conf", true, utilsHelpers)
+	if err != nil {
+		return fmt.Errorf("failed to save NFT ruleset to file on node %s: %w", nodeName, err)
+	}
+
+	_, err = runBashCommandOnPod(debugPod, "systemctl enable nftables", true, utilsHelpers)
+	if err != nil {
+		return fmt.Errorf("failed to enable nftables on node %s: %w", nodeName, err)
+	}
+
+	err = saveNFTablesRules("commatrix", "nftables-"+nodeName, string(output))
```

Comment:
I don't understand why you need this function here instead of just printing to file after generating output here https://github.com/openshift-kni/commatrix/pull/36/files#diff-2d8a2bd5eebac03f17584c31940f9f92f6353546ec5b4f33b494045cf94fecdcR41

Reply:
it will be so much lines